### PR TITLE
Allow for mocking with Jasmine

### PR DESCRIPTION
**Pull request by ElvisASC4**

### CODE REVIEW

File: ui/src/AudioRecorder.js [hunk 1]
```diff
@@ -3,28 +3,45 @@ export class AudioRecorder {
     * Creates an AudioRecorder that can record utterances
```
Comment (ashleygranitto) [inline ui/src/AudioRecorder.js:3]: Just a heads up that there will probably be many merge conflicts with this file (I believe Eldrick & I both made changes) so let us know if you need any help merging all of the versions

File: ui/src/StubMediaRecorder.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export class StubMediaRecorder {
```
Comment (ashleygranitto) [inline ui/src/StubMediaRecorder.js:1]: @everyone--Going forward for testing, should we make a new file for each stub?

File: ui/src/VoxettaRecordButton.js [hunk 3]
```diff
@@ -38,11 +38,13 @@ export class VoxettaRecordButton extends LitElement {
      async recordHandler(e){
```
Comment (ashleygranitto) [inline ui/src/VoxettaRecordButton.js:38]: I think e can be removed without issue

File: ui/jasmine_test/spec/test.spec.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import {StubMediaRecorder} from '../../src/StubMediaRecorder.js';
```
Comment (ashleygranitto) [inline ui/jasmine_test/spec/test.spec.js:1]: this file looks so clean *sunglasses emoji*

File: ui/jasmine_test/spec/helpers/jsdom_init.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+// var jsdom = require("jsdom");
```
Comment (win-chen) [inline ui/jasmine_test/spec/helpers/jsdom_init.js:1]: remove commented out code

File: ui/jasmine_test/spec/helpers/jsdom_init.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+import {JSDOM} from 'jsdom';
+import {StubMediaRecorder} from '../../../src/StubMediaRecorder.js';
+import {StubSoundWave} from '../../../src/StubSoundWave.js';
```
Comment (win-chen) [inline ui/jasmine_test/spec/helpers/jsdom_init.js:4]: StubSoundWave is not being used in this file, remove

File: ui/jasmine_test/spec/helpers/jsdom_init.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+// var jsdom = require("jsdom");
+import {JSDOM} from 'jsdom';
+import {StubMediaRecorder} from '../../../src/StubMediaRecorder.js';
```
Comment (win-chen) [inline ui/jasmine_test/spec/helpers/jsdom_init.js:3]: I'd suggest moving the StubMediaRecorder.js file into a spec/helpers/stubs folder

File: ui/jasmine_test/spec/test.spec.js (hunk 4 above)
Comment (win-chen) [inline ui/jasmine_test/spec/test.spec.js:1]: remove unused import {StubMediaRecorder}

File: ui/jasmine_test/spec/test.spec.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+    });
+    
+    it('records audio', async () => {
+        const audioRecorder = new AudioRecorder();
+        await audioRecorder.initRecorder();
+        audioRecorder.startRecording();
+        expect(audioRecorder.mediaRecorder.start).toHaveBeenCalled();
+    });
+
+    it('Stops recording', async () => {
```
Comment (win-chen) [inline ui/jasmine_test/spec/test.spec.js:27]: nit: be consistent with your capitalization

File: ui/package.json [hunk 9]
```diff
@@ -9,13 +9,17 @@
...
         "start:build": "npm run build && es-dev-server --root-dir dist --app-index index.html --compatibility none --open",
         "format": "prettier src/* --write",
-        "lint": "eslint src/*"
+        "lint": "eslint src/*",
+        "test": "babel-node jasmine_test/spec/run.js"
     },
     "dependencies": {
-        "lit-html": "^1.0.0",
-        "lit-element": "^2.0.1"
+        "babel-cli": "^6.26.0",
```
Comment (win-chen) [inline ui/package.json:16]: this should probably be a dev dependecy

File: ui/jasmine_test/spec/helpers/jsdom_init.js (hunk 7 above)
Comment (mmallett) [reply ui/jasmine_test/spec/helpers/jsdom_init.js:3]: +1

File: ui/src/StubMediaRecorder.js (hunk 2 above)
Comment (mmallett) [reply ui/src/StubMediaRecorder.js:1]: Yeah, create a new file for each stub. As Winnie commented above, stubs should go in `ui/jasmine_test/spec/helpers`

File: ui/src/VoxettaRecordButton.js (hunk 3 above)
Comment (ElvisASC4) [reply ui/src/VoxettaRecordButton.js:38]: I'll fix that in the other branch

**Pull request merged**